# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/nr_socket_proxy_config.h
# Commit: f5353ea9f1df
# Full Hash: f5353ea9f1dfe15a65076bc7aa272d688ebdb8ca
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Add a proxy policy argument to NrSocketProxyConfig, and always set this config on the NrIceCtx. r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45101
# ==============================================================================

diff -r ed1ae6a85255 -r f5353ea9f1df media/mtransport/nr_socket_proxy_config.h
--- a/media/mtransport/nr_socket_proxy_config.h	Tue Sep 17 18:17:13 2019 +0000
+++ b/media/mtransport/nr_socket_proxy_config.h	Tue Sep 17 18:17:14 2019 +0000
@@ -19,8 +19,11 @@
 
 class NrSocketProxyConfig {
  public:
+  typedef enum { kDisableProxy, kEnableProxy, kForceProxy } ProxyPolicy;
+
   NrSocketProxyConfig(uint64_t aTabId, const nsCString& aAlpn,
-                      const net::LoadInfoArgs& aArgs);
+                      const net::LoadInfoArgs& aArgs, ProxyPolicy aProxyPolicy);
+
   // We need to actually write the default impl ourselves, because the compiler
   // needs to know how to destroy mPrivate in case an exception is thrown, even
   // though we disable exceptions in our build.
@@ -31,6 +34,7 @@
   uint64_t GetTabId() const;
   const nsCString& GetAlpn() const;
   const net::LoadInfoArgs& GetLoadInfoArgs() const;
+  ProxyPolicy GetProxyPolicy() const;
 
  private:
   // LoadInfoArgs includes stuff that conflicts with nICEr includes.